### PR TITLE
Add NewTimer to util.Clock interface

### LINKED ISSUE #25953: Unit test flake: TestReflectorResync exactly 2 iterations were expected, got: 3
```
-- FAIL: TestReflectorResync (0.01s)
	reflector_test.go:380: exactly 2 iterations were expected, got: 3
```
https://console.cloud.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/25904/kubernetes-pull-test-unit-integration/26834/

### PR DESCRIPTION
**Pull request by ncdc**
Add NewTimer to util.Clock - Add Timer interface to allow testing timers in a controlled manner, for testing purposes.

Make reflector use util.Timer interface.

Fixes #25953

### CODE REVIEW

**Comment by ncdc** [comment]
@smarterclayton @kubernetes/rh-cluster-infra @lavalamp @liggitt @mml

**Comment by ncdc** [comment]
@k8s-bot node e2e test this issue: #25974

File: pkg/client/cache/reflector_test.go [hunk 1]
```diff
@@ -375,7 +376,24 @@ func TestReflectorResync(t *testing.T) {
...
 	resyncPeriod := 1 * time.Millisecond
 	r := NewReflector(lw, &api.Pod{}, s, resyncPeriod)
-	r.ListAndWatch(stopCh)
+	clock := util.NewFakeClock(time.Now())
+	r.clock = clock
+
+	go r.ListAndWatch(stopCh)
+
+	// advance the clock until the reflector stops
+Loop:
```
Comment (xiang90) [inline pkg/client/cache/reflector_test.go]: I do not think this solves the problem. If resyncfunc hits line 362 while this for loop is at line 393, the timer will still be increased at line 387. Then listAndWatch might call resyncfunc again.

Comment (xiang90) [inline pkg/client/cache/reflector_test.go]: @ncdc https://github.com/kubernetes/kubernetes/pull/26013 this simple fix might work.

**Comment by wojtek-t** [comment]
@ncdc - the bug you are trying to fix should be fixed by #26013. That particular PR is better in my opinion. So I'm going to proceed with that one and close this one.

If because of some reason, you want this PR anyway, please reopen it.

**Comment by ncdc** [comment]
@wojtek-t no objections. Do you think the Timer/NewTimer commit is worthwhile in util.Clock?

**Comment by wojtek-t** [comment]
@ncdc - for now I don't see any gains from having it. So maybe let's get back to this PR if we find some usecases for it.

**Pull request closed without merging**